### ISSUE TITLE
TestActivityRetryOnHBTimeout is flaky

### ISSUE THREAD

**Issue by vitarb** [temporalio/sdk-go#546]
Error from the build pipeline:
```
=== RUN   TestIntegrationSuite/TestActivityRetryOnHBTimeout

Handling connection for 7233

2021/09/23 07:46:13 INFO  Started Worker Namespace integration-test-namespace TaskQueue tq-5-TestIntegrationSuite/TestActivityRetryOnHBTimeout WorkerID 6171@fdbd4600-e1aa-46fb-6b79-777ba48942e4@

2021/09/23 07:46:13 DEBUG ExecuteActivity Namespace integration-test-namespace TaskQueue tq-5-TestIntegrationSuite/TestActivityRetryOnHBTimeout WorkerID 6171@fdbd4600-e1aa-46fb-6b79-777ba48942e4@ WorkflowType ActivityRetryOnHBTimeout WorkflowID test-activity-retry-on-hbtimeout RunID 362e659a-3ea4-4d68-9577-461d4230c1f7 Attempt 1 ActivityID 5 ActivityType HeartbeatAndSleep

2021/09/23 07:46:13 INFO  Running HeartbeatAndSleep activity Namespace integration-test-namespace TaskQueue tq-5-TestIntegrationSuite/TestActivityRetryOnHBTimeout WorkerID 6171@fdbd4600-e1aa-46fb-6b79-777ba48942e4@ ActivityID 5 ActivityType HeartbeatAndSleep Attempt 1 WorkflowType ActivityRetryOnHBTimeout WorkflowID test-activity-retry-on-hbtimeout RunID 362e659a-3ea4-4d68-9577-461d4230c1f7

2021/09/23 07:46:15 INFO  Task processing failed with error Namespace integration-test-namespace TaskQueue tq-5-TestIntegrationSuite/TestActivityRetryOnHBTimeout WorkerID 6171@fdbd4600-e1aa-46fb-6b79-777ba48942e4@ WorkerType ActivityWorker Error invalid activityID or activity already timed out or invoking workflow is completed

2021/09/23 07:46:16 INFO  Running HeartbeatAndSleep activity Namespace integration-test-namespace TaskQueue tq-5-TestIntegrationSuite/TestActivityRetryOnHBTimeout WorkerID 6171@fdbd4600-e1aa-46fb-6b79-777ba48942e4@ ActivityID 5 ActivityType HeartbeatAndSleep Attempt 2 WorkflowType ActivityRetryOnHBTimeout WorkflowID test-activity-retry-on-hbtimeout RunID 362e659a-3ea4-4d68-9577-461d4230c1f7

2021/09/23 07:46:18 INFO  Task processing failed with error Namespace integration-test-namespace TaskQueue tq-5-TestIntegrationSuite/TestActivityRetryOnHBTimeout WorkerID 6171@fdbd4600-e1aa-46fb-6b79-777ba48942e4@ WorkerType ActivityWorker Error invalid activityID or activity already timed out or invoking workflow is completed

2021/09/23 07:46:19 INFO  Running HeartbeatAndSleep activity Namespace integration-test-namespace TaskQueue tq-5-TestIntegrationSuite/TestActivityRetryOnHBTimeout WorkerID 6171@fdbd4600-e1aa-46fb-6b79-777ba48942e4@ ActivityID 5 ActivityType HeartbeatAndSleep Attempt 3 WorkflowType ActivityRetryOnHBTimeout WorkflowID test-activity-retry-on-hbtimeout RunID 362e659a-3ea4-4d68-9577-461d4230c1f7

    TestIntegrationSuite: integration_test.go:382: 

        	Error Trace:	integration_test.go:382

        	Error:      	Received unexpected error:

        	            	workflow execution error (type: ActivityRetryOnHBTimeout, workflowID: test-activity-retry-on-hbtimeout, runID: 362e659a-3ea4-4d68-9577-461d4230c1f7): expected activity to fail but succeeded

        	Test:       	TestIntegrationSuite

2021/09/23 07:46:21 INFO  Stopped Worker Namespace integration-test-namespace TaskQueue tq-5-TestIntegrationSuite/TestActivityRetryOnHBTimeout WorkerID 6171@fdbd4600-e1aa-46fb-6b79-777ba48942e4@

    TestIntegrationSuite/TestActivityRetryOnHBTimeout: testing.go:916: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
```

**Comment by cretz** [comment]
I think this was fixed (or at least improved) by #545

**Comment by cretz** [comment]
This test does not appear flaky anymore in CI
